syntax = "proto3";
package syft.core.io;

import "proto/core/common/common_object.proto";
import "proto/core/io/location.proto";
import "proto/core/io/connection.proto";
import "proto/grid/connections/http_connection.proto";

message SoloRoute {
  syft.core.common.UID id = 1;
  syft.core.io.SpecificLocation destination = 2;
  oneof connection {
    syft.core.io.VirtualClientConnection virtual_connection = 3;
    syft.grid.connections.GridHTTPConnection grid_connection = 4;
  }
}